src/pools/pool-internal.ts

```diff
@@ -2,6 +2,14 @@ import EventEmitter from 'events'
 import type { IWorker } from './abstract-pool'
 import type { IPool } from './pool'
 
+/**
+ * Pool types.
+ */
+export enum PoolType {
```

Comment:
As long as we do not export this to public, this can be an TS enum.  
Otherwise we should use `Object.freeze` as the other _enums_.

Reply:
Any chance to use Object.freeze here?

Reply:
I've not really well understood why @Shinigami92 changed TS enum in the first place. 
The enum transpiled code looks really well optimised to me with TS 4.2.x, so I see no gain on using `const`, `Object.freeze`, etc.

Reply:
@jerome-benoit Could you provide me with a resource to this 4.2 optimization? Maybe I missed something new :thinking:  

But @pioardi: this is only used internally and therefore it is okay as long it is not exported to public (e.g. via `index.ts`)

Reply:
I do not follow closely all changes in TS release. All I can say here is that the transpiled code for that enum is: 

```js
exports.PoolType = void 0;
(function (PoolType) {
    PoolType["FIXED"] = "fixed";
    PoolType["DYNAMIC"] = "dynamic";
})(exports.PoolType || (exports.PoolType = {}));
```

which looks pretty efficient to me.

Reply:
`Object.freeze` doesn't get transpiled at all and are directly native JS  
Are JSDoc preserved on the enum values also, so JS users that don't use TS also see them?

Reply:
Used something to read I just quickly googled: https://blog.logrocket.com/why-typescript-enums-suck

tl;dr as long as it is just for internal usage it is okay IMO
But otherwise, please try to rely on more native JS
Users can then also use simply e.g. `'HARD'` instead of `KillBehaviors.HARD`

Reply:
Fair enough. 

So that makes that PR mergeable? ;-)

Reply:
From my side: Yes :+1: